### DISCUSSION TITLE
Check if form fields are present in form state

### DISCUSSION THREAD

**Discussion by zoltanr-jt** [react-hook-form/react-hook-form#9565] [Ideas]
Hey,
I have a suggestion for a new feature for check if all initially defined form keys are in the form state.

Validation currently validated the form values, but would be great to have validation if all the form inputs - registered and unregistered are present in the form state.

This type of validation would come handy when form is used in wizard and separately rendered panels
When a input is not rendered it is not registered on the form state -> but the form is still considered valid.

In more detail:
Let's say we have a form with the following initial state:

```
{
 name: undefined,
 age: undefined
}
```

Now let's assume that the form inputs  of the above states are are registered via the context and they rendering in  separate tab of a step-by-step wizard.

Wizard panels
- step 1: input for `name`
- step 2 input for `age`
- step 3: step with a preview of form state and submit button

The problem:fi
then the wizard renders the first panel (step 1) it renders the input for the `name` field, and the field gets registered on the  form state .
IF the user skips the 2nd panel and jumps to step 3, the form state is valid but it does not contain the `age` field because the input was not rendered and this not registered.

Would be nice to have feature to add a flag to check for all the mandatory field if they are present on the global form state.

**Comment by bluebill1049** [comment]
Do you mean expose registered inputs?

**Reply by KarafiziArtur** [reply]
Hi! We also encountered a case when we need to see all registered fields that exist to show a specific error if it doesn't exist. Our form is complicated and we need to manage this case as well
I can't find this opportunity to see all registered fields in docs
